### DISCUSSION TITLE
I wonder why support for the media used in UserMessage is deprecated.

### DISCUSSION THREAD

**Discussion by yeseong0412** [spring-projects/spring-ai#1274] [Q&A] [answered]
https://github.com/spring-projects/spring-ai/blob/main/spring-ai-core/src/main/java/org/springframework/ai/model/Media.java

Currently, UserMessage is created with List<Media> mediaList as the [constructor](https://github.com/spring-projects/spring-ai/blob/e1884d1d92518aabb8a3fdb86e513b898ff79527/spring-ai-core/src/main/java/org/springframework/ai/chat/messages/UserMessage.java#L47). However, if [Media](https://github.com/spring-projects/spring-ai/blob/e1884d1d92518aabb8a3fdb86e513b898ff79527/spring-ai-core/src/main/java/org/springframework/ai/model/Media.java#L48) has been discontinued since version 1.0.0 M1, what can the current Media be replaced with?

++ Is there already a new class to replace media?

**Comment by ThomasVitale** [comment, marked as answer]
The deprecation affects only the constructor taking an `Object` as input for building the `Media`. You can still build a `Media` object by passing a `org.springframework.core.io.Resource` or a URL.

https://github.com/spring-projects/spring-ai/blob/main/spring-ai-core/src/main/java/org/springframework/ai/model/Media.java#L40-L70

**Reply by ThomasVitale** [reply]
Example of both cases: https://github.com/ThomasVitale/llm-apps-java-spring-ai/blob/main/04-multimodality/multimodality-openai/src/main/java/com/thomasvitale/ai/spring/model/ChatModelService.java#L30

**Reply by yeseong0412** [reply]
> The deprecation affects only the constructor taking an `Object` as input for building the `Media`. You can still build a `Media` object by passing a `org.springframework.core.io.Resource` or a URL.
> 
> https://github.com/spring-projects/spring-ai/blob/main/spring-ai-core/src/main/java/org/springframework/ai/model/Media.java#L40-L70

Oh.. Thank you
